[sc-435] Add support for custom entrypoints (#182)
HigherOrderCO/hvm-lang: docs/compiler-options.md

| `-Osimplify-main` `-Ono-simplify-main` | Disabled | [simplify-main](#simplify-main) |
| `-Omerge` `-Ono-merge` | Disabled | [definition-merging](#definition-merging) |
| `-Oinline` `-Ono-inline` | Disabled | [inline](#inline) |

> | `-e` `--entrypoint` | `Main \| main` | Use another definition as entrypoint, rather than `main` or `Main`. |

Reviewer: I think it's best to expand this into it's own section so that we can explain the compiled output difference between having a custom entry point and not having one.

@main = a
& (b b) ~ ([#2 #3] a)
```

## Entrypoint

If given the option, use another definition as entrypoint rather than `main` or `Main`.

> HVM-Lang finds by default for a `main` or `Main` as entrypoint to a program, but it is possible to use a different entrypoint with the `-e --entrypoint` option.

Example:
```
// program
Main = (λx x λx x)

> // compilation output, Main or main becomes a @main definition

Reviewer: ```suggestion
// compilation output, both `Main` and `main` compile to a `@main` definition
```

@main = a
& (b b) ~ ((c c) a)

// program

> // should fail as the program does not have a `main` definition.

Reviewer: ```suggestion
// Normally would fail as the program does not have a `main` definition.
```

run = (λx x λx x)

> // compilation output, --entrypoint run.

Reviewer: ```suggestion
// compilation output using `--entrypoint run`.
// No `@main` definition is created.
```
